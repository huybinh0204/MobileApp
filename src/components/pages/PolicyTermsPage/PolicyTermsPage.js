import { useNavigation } from '@react-navigation/native';
import React, { useRef } from 'react';
import { IconSvg } from '_components/atoms';
import { PressableList } from '_components/molecules';
import { SecondaryHeader } from '_components/organisms';
import { ICONS, NAVIGATION } from '_constants';
import strings from '_localization/index';
import { Container, listStyles } from './PolicyTermsPage.styles';

const arrowIcon = (props) => <IconSvg icon={ICONS.arrowRight} {...props} />;

const PolicyTermsPage = () => {
  const { navigate } = useNavigation();

  const { current: items } = useRef([
    {
      title: strings.settings.policyTerms.depositAccount.title,
      rightItem: arrowIcon,
      testID: 'PolicyDepositAccountAgreementOption',
      onPress: () => {
        navigate(NAVIGATION.settings.accountPolicyDetails, {
          url: strings.settings.policyTerms.depositAccount.url,
        });
      },
    },
    {
      title: strings.settings.policyTerms.electronicCommunication.title,
      rightItem: arrowIcon,
      testID: 'PolicyDepositAccountAgreementOption',
      onPress: () => {
        navigate(NAVIGATION.settings.accountPolicyDetails, {
          url: strings.settings.policyTerms.electronicCommunication.url,
        });
      },
    },
    {
      title: strings.settings.policyTerms.privacyPolicy.title,
      rightItem: arrowIcon,
      testID: 'PolicyDepositAccountAgreementOption',
      onPress: () => {
        navigate(NAVIGATION.settings.accountPolicyDetails, {
          url: strings.settings.policyTerms.privacyPolicy.url,
        });
      },
    },
    {
      title: strings.settings.policyTerms.kinlyUser.title,
      rightItem: arrowIcon,
      testID: 'PolicyDepositAccountAgreementOption',
      onPress: () => {
        navigate(NAVIGATION.settings.accountPolicyDetails, {
          url: strings.settings.policyTerms.kinlyUser.url,
        });
      },
    },
    {
      title: strings.settings.policyTerms.overdraft.title,
      rightItem: arrowIcon,
      testID: 'PolicyDepositAccountAgreementOption',
      onPress: () => {
        navigate(NAVIGATION.settings.accountPolicyDetails, {
          url: strings.settings.policyTerms.overdraft.url,
        });
      },
    },
    {
      title: strings.settings.policyTerms.bankPrivacy.title,
      rightItem: arrowIcon,
      testID: 'PolicyDepositAccountAgreementOption',
      onPress: () => {
        navigate(NAVIGATION.settings.accountPolicyDetails, {
          url: strings.settings.policyTerms.bankPrivacy.url,
        });
      },
    },
  ]);

  return (
    <Container testID="PolicyTermsPage">
      <SecondaryHeader title={strings.settings.account_policy} />
      <PressableList items={items} contentContainerStyle={listStyles} />
    </Container>
  );
};

export default PolicyTermsPage;
