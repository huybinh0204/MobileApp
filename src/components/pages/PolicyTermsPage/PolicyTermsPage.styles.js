import styled, { css } from '@emotion/native';
import Spacing from '_constants/Spacing';

export const Container = styled.SafeAreaView(({ theme }) => ({
  flex: 1,
  backgroundColor: theme.colors.white,
}));

export const listStyles = css({
  paddingHorizontal: Spacing.m,
});
