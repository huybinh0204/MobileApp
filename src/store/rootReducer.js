import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
