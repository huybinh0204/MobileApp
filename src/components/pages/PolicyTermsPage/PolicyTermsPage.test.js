import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import { createStore } from 'redux';
import PolicyTermsPage from './PolicyTermsPage';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';

describe('PolicyTermsPage', () => {
  it('should match snapshot', async () => {
    const mockStore = createStore(rootReducer, {});

    const { toJSON } = render(
      <ComponentWithProviders component={PolicyTermsPage} store={mockStore} />
    );

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
