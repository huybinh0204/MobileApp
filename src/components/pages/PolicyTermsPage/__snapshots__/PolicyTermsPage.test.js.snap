// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyTermsPage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="PolicyTermsPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#1F1D25",
                                  "fontFamily": "Poppins-SemiBold",
                                  "fontSize": 29,
                                  "lineHeight": 40,
                                },
                                undefined,
                              ]
                            }
                          >
                            Account Policy & Terms
                          </Text>
                        </View>
                        <RCTScrollView
                          ItemSeparatorComponent={[Function]}
                          contentContainerStyle={
                            Object {
                              "paddingHorizontal": 58,
                            }
                          }
                          data={
                            Array [
                              Object {
                                "onPress": [Function],
                                "rightItem": [Function],
                                "testID": "PolicyDepositAccountAgreementOption",
                                "title": "Ahead Deposit Account Agreement",
                              },
                              Object {
                                "onPress": [Function],
                                "rightItem": [Function],
                                "testID": "PolicyDepositAccountAgreementOption",
                                "title": "Ahead Agreement to Receive Electronic Communications",
                              },
                              Object {
                                "onPress": [Function],
                                "rightItem": [Function],
                                "testID": "PolicyDepositAccountAgreementOption",
                                "title": "Ahead Privacy Policy",
                              },
                              Object {
                                "onPress": [Function],
                                "rightItem": [Function],
                                "testID": "PolicyDepositAccountAgreementOption",
                                "title": "Ahead User Agreement",
                              },
                              Object {
                                "onPress": [Function],
                                "rightItem": [Function],
                                "testID": "PolicyDepositAccountAgreementOption",
                                "title": "Ahead Overdraft Terms and Conditions",
                              },
                              Object {
                                "onPress": [Function],
                                "rightItem": [Function],
                                "testID": "PolicyDepositAccountAgreementOption",
                                "title": "The Bancorp Bank Privacy Notice",
                              },
                            ]
                          }
                          getItem={[Function]}
                          getItemCount={[Function]}
                          keyExtractor={[Function]}
                          onContentSizeChange={[Function]}
                          onLayout={[Function]}
                          onMomentumScrollBegin={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          removeClippedSubviews={false}
                          renderItem={[Function]}
                          scrollEventThrottle={50}
                          stickyHeaderIndices={Array []}
                          viewabilityConfigCallbackPairs={Array []}
                        >
                          <View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "flexDirection": "row",
                                    "minHeight": 50,
                                    "opacity": 1,
                                  }
                                }
                                testID="PolicyDepositAccountAgreementOption"
                              >
                                <Text
                                  allowFontScaling={false}
                                  marginLeft={false}
                                  style={
                                    Array [
                                      Object {
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "marginLeft": 0,
                                      },
                                      null,
                                    ]
                                  }
                                  testID="ItemTitle"
                                >
                                  Ahead Deposit Account Agreement
                                </Text>
                                <SvgMock
                                  height={45.5}
                                  testID="RightItem"
                                  width={45.5}
                                />
                              </View>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "borderColor": "#F2F3F4",
                                      "borderWidth": 1,
                                      "height": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "flexDirection": "row",
                                    "minHeight": 50,
                                    "opacity": 1,
                                  }
                                }
                                testID="PolicyDepositAccountAgreementOption"
                              >
                                <Text
                                  allowFontScaling={false}
                                  marginLeft={false}
                                  style={
                                    Array [
                                      Object {
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "marginLeft": 0,
                                      },
                                      null,
                                    ]
                                  }
                                  testID="ItemTitle"
                                >
                                  Ahead Agreement to Receive Electronic Communications
                                </Text>
                                <SvgMock
                                  height={45.5}
                                  testID="RightItem"
                                  width={45.5}
                                />
                              </View>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "borderColor": "#F2F3F4",
                                      "borderWidth": 1,
                                      "height": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "flexDirection": "row",
                                    "minHeight": 50,
                                    "opacity": 1,
                                  }
                                }
                                testID="PolicyDepositAccountAgreementOption"
                              >
                                <Text
                                  allowFontScaling={false}
                                  marginLeft={false}
                                  style={
                                    Array [
                                      Object {
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "marginLeft": 0,
                                      },
                                      null,
                                    ]
                                  }
                                  testID="ItemTitle"
                                >
                                  Ahead Privacy Policy
                                </Text>
                                <SvgMock
                                  height={45.5}
                                  testID="RightItem"
                                  width={45.5}
                                />
                              </View>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "borderColor": "#F2F3F4",
                                      "borderWidth": 1,
                                      "height": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "flexDirection": "row",
                                    "minHeight": 50,
                                    "opacity": 1,
                                  }
                                }
                                testID="PolicyDepositAccountAgreementOption"
                              >
                                <Text
                                  allowFontScaling={false}
                                  marginLeft={false}
                                  style={
                                    Array [
                                      Object {
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "marginLeft": 0,
                                      },
                                      null,
                                    ]
                                  }
                                  testID="ItemTitle"
                                >
                                  Ahead User Agreement
                                </Text>
                                <SvgMock
                                  height={45.5}
                                  testID="RightItem"
                                  width={45.5}
                                />
                              </View>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "borderColor": "#F2F3F4",
                                      "borderWidth": 1,
                                      "height": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "flexDirection": "row",
                                    "minHeight": 50,
                                    "opacity": 1,
                                  }
                                }
                                testID="PolicyDepositAccountAgreementOption"
                              >
                                <Text
                                  allowFontScaling={false}
                                  marginLeft={false}
                                  style={
                                    Array [
                                      Object {
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "marginLeft": 0,
                                      },
                                      null,
                                    ]
                                  }
                                  testID="ItemTitle"
                                >
                                  Ahead Overdraft Terms and Conditions
                                </Text>
                                <SvgMock
                                  height={45.5}
                                  testID="RightItem"
                                  width={45.5}
                                />
                              </View>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "borderColor": "#F2F3F4",
                                      "borderWidth": 1,
                                      "height": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "flexDirection": "row",
                                    "minHeight": 50,
                                    "opacity": 1,
                                  }
                                }
                                testID="PolicyDepositAccountAgreementOption"
                              >
                                <Text
                                  allowFontScaling={false}
                                  marginLeft={false}
                                  style={
                                    Array [
                                      Object {
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "marginLeft": 0,
                                      },
                                      null,
                                    ]
                                  }
                                  testID="ItemTitle"
                                >
                                  The Bancorp Bank Privacy Notice
                                </Text>
                                <SvgMock
                                  height={45.5}
                                  testID="RightItem"
                                  width={45.5}
                                />
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
